sabnzbd/utils/correct_extension.py

```diff
@@ -0,0 +1,316 @@
+    "cgi",
+    "cgi",
+    "cgi",
+    "class",
+    "com",
+    "cpl",
+    "cpp",
+    "cs",
+    "css",
+    "csv",
+    "cur",
+    "dat",
+    "db",
+    "dbf",
+    "deb",
+    "dll",
+    "dmg",
+    "dmp",
+    "doc",
+    "docx",
+    "drv",
+    "email",
+    "eml",
+    "emlx",
+    "exe",
+    "flv",
+    "fnt",
+    "fon",
+    "gadget",
+    "gif",
+    "h",
+    "h264",
+    "htm",
+    "html",
+    "icns",
+    "ico",
+    "ico",
+    "ini",
+    "iso",
+    "jar",
+    "java",
+    "jpeg",
+    "jpg",
+    "js",
+    "jsp",
+    "key",
+    "lnk",
+    "log",
+    "m4v",
+    "mdb",
+    "mid",
+    "midi",
+    "mkv",
+    "mov",
+    "mp3",
+    "mp4",
+    "mpa",
+    "mpeg",
+    "mpg",
+    "msg",
+    "msi",
+    "msi",
+    "odp",
+    "ods",
+    "odt",
+    "oft",
+    "ogg",
+    "ost",
+    "otf",
+    "part",
+    "pdf",
+    "php",
+    "php",
+    "pkg",
+    "pl",
+    "pl",
+    "pl",
+    "png",
+    "pps",
+    "ppt",
+    "pptx",
+    "ps",
+    "psd",
+    "pst",
+    "py",
+    "py",
+    "py",
+    "rar",
+    "rm",
+    "rpm",
+    "rss",
+    "rtf",
+    "sav",
+    "sh",
+    "sql",
+    "svg",
+    "swf",
+    "swift",
+    "sys",
+    "tar",
+    "tar",
+    "gz",
+    "tex",
+    "tif",
+    "tiff",
+    "tmp",
+    "toast",
+    "ttf",
+    "txt",
+    "vb",
+    "vcd",
+    "vcf",
+    "vob",
+    "wav",
+    "wma",
+    "wmv",
+    "wpd",
+    "wpl",
+    "wsf",
+    "xhtml",
+    "xls",
+    "xlsm",
+    "xlsx",
+    "xml",
+    "z",
+    "zip",
+)
+
+DOWNLOAD_EXT = (
+    "ass",
+    "avi",
+    "bat",
+    "bdmv",
+    "bin",
+    "bup",
+    "clpi",
+    "crx",
+    "db",
+    "diz",
+    "docx",
+    "epub",
+    "exe",
+    "flac",
+    "gif",
+    "gz",
+    "htm",
+    "html",
+    "icns",
+    "ico",
+    "idx",
+    "ifo",
+    "img",
+    "inf",
+    "info",
+    "ini",
+    "iso",
+    "jpg",
+    "log",
+    "m2ts",
+    "m3u",
+    "m4a",
+    "mkv",
+    "mp3",
+    "mp4",
+    "mpls",
+    "mx",
+    "nfo",
+    "nib",
+    "nzb",
+    "otf",
+    "par2",
+    "part",
+    "pdf",
+    "pem",
+    "php",
+    "plist",
+    "png",
+    "py",
+    "rar",
+    "releaseinfo",
+    "rev",
+    "sfv",
+    "sh",
+    "srr",
+    "srs",
+    "srt",
+    "strings",
+    "sub",
+    "sup",
+    "sys",
+    "tif",
+    "ttf",
+    "txt",
+    "url",
+    "vob",
+    "website",
+    "wmv",
+    "xml",
+    "xpi",
+)
+
+RAR_EXT = tuple(["r" + str(i).zfill(2) for i in range(0, 100)])  # "r00", "r01", ...
+
+# combine to one tuple, with unique entries:
+ALL_EXT = tuple(set(COMMON_EXT + DOWNLOAD_EXT + RAR_EXT))
+
+
+def has_common_extension(file_path: str) -> int:
+    """returns boolean if the extension of file_path is a common, well-known extension"""
+    # TBD use SAB's own extension finder
+    filename, file_extension = os.path.splitext(file_path)
+    file_extension = file_extension[1:].lower()
+    return file_extension in ALL_EXT
+
+
+def all_possible_extensions(file_path: str) -> list:
```

Comment:
Use `List[puremagic.PureMagicWithConfidence]` 
See here for diference between `list` and `List`:
https://stackoverflow.com/questions/39458193/using-list-tuple-etc-from-typing-vs-directly-referring-type-as-list-tuple-etc

Reply:
typing.List works, but

```
>>> typing.List[puremagic.PureMagicWithConfidence]
/home/sander/.local/lib/python3.8/site-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.25.9) or chardet (4.0.0) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'puremagic' has no attribute 'PureMagicWithConfidence'
```
FWIW

```
>>> puremagic.
puremagic.PureError(          puremagic.from_file(          puremagic.from_string(        puremagic.magic_footer_array  puremagic.magic_stream(       puremagic.main                
puremagic.ext_from_filename(  puremagic.from_stream(        puremagic.magic_file(         puremagic.magic_header_array  puremagic.magic_string(    
```

Reply:
@Safihre ... can you help?

Reply:
@sanderjo looks like you don't need that `PureMagicWithConfidence` type: the returned list contains only the extensions - and those are just regular `str`.

Reply:
True!
But it should still be `List[str]` and not `list`